readthedocs/search/tests/test_api.py

```diff
@@ -439,6 +439,43 @@ def test_search_correct_link_for_index_page_subdirectory_htmldir_projects(self,
         assert result['project'] == project.slug
         assert result['link'].endswith('en/latest/guides/')
 
+    def test_search_advanced_query_detection(self, api_client):
+        project = Project.objects.get(slug='docs')
+        version = project.versions.all().first()
+
+        # Query with a typo should return results
+        search_params = {
+            'project': project.slug,
+            'version': version.slug,
+            'q': 'indx',
+        }
+        resp = self.get_search(api_client, search_params)
+        assert resp.status_code == 200
+
+        assert len(resp.data['results']) > 0
```

Comment:
We should probably validate it returns the result we expect (`index in resp.data['results']` or something)